crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs

```diff
@@ -273,6 +317,42 @@ impl RestrictedPattern {
 
         GlobResult::Found
     }
+
+    fn get_regex_result(&self, name: &NameSpan) -> bool {
+        let Some(regex) = &self.regex else {
+            return false;
+        };
+
+        let Ok(reg_exp) = Regex::new(regex.as_str()) else {
```

Comment:
probably a TODO - use https://doc.rust-lang.org/std/sync/struct.OnceLock.html to avoid having to re-compile the regex a bunch of times

Reply:
I do not know how to use OnceLock with an array of `patterns` and inside multiple regex-options.
I reduced the call with this commit: https://github.com/oxc-project/oxc/pull/8076/commits/87d8da7f18ecfe3f403ef03a85b6092ee85194ed

@camc314  Hope this helps ❤️